prettyconf/loaders.py

```diff
@@ -20,36 +22,102 @@ def __getitem__(self, item):
         raise NotImplementedError()  # pragma: no cover
 
 
-class EnvVarConfigurationLoader(AbstractConfigurationLoader):
+class CommandLine(AbstractConfigurationLoader):
+    """
+    Works with `argparse` parsers.
+    """
+    default = '==PRETTYCONF=='
+    def __init__(self, parser):
+        # We have to put a special default value so that we know if the
+        # variable has been set by the user explicitly.
+        _parser = ArgumentParser()
+        for action in parser._actions:
+            copy = Action(**dict(action._get_kwargs()))
+            copy.default = self.default
+            _parser._actions.append(copy)
+        self.configs = _parser.parse_args()
+
     def __contains__(self, item):
-        return item in os.environ
+        return item in self.configs and getattr(self.configs, item) != self.default
 
     def __getitem__(self, item):
-        return os.environ[item]
+        try:
+            res = getattr(self.configs, item)
+        except AttributeError:
+            raise KeyError("{!r}".format(item))
+
+        if res == self.default:
+            raise KeyError("{!r}".format(item))
+        return res
 
 
-class AbstractFileConfigurationLoader(AbstractConfigurationLoader):
-    patterns = ()
+class IniFile(AbstractConfigurationLoader):
```

Comment:
Try to not change the order of class definitions to make patch smaller and easier to review...

Reply:
Sorry